File: python/_queue.py
class Queue:
    def __init__(self) -> None:
        self.queue = []
        self.front = 0
        self.back = -1
        
    def enqueue(self, item: str):
        self.back += 1
        self.queue[self.back] = item
        
    def dequeue(self):
        self.front += 1
        if self.front > self.back:
            raise Exception("Queue underflow")
        
    def peek(self):
        return self.queue[self.front]

File: js/queue.js
class Queue{
    constructor() {
        this.queue = []
        this.front = 0
        this.back = -1
    }

    enqueue(item){
        this.back++
        this.queue[this.back] = item
    }

    dequeue(){
        this.front++
        if(this.front > this.back){
            throw new Error("Queue underflow")
        }
    }

    peek(){
        return this.queue[this.front]
    }
}

module.exports = Queue
